src/Autofac/Core/Container.cs

```diff
@@ -100,10 +100,7 @@ public ILifetimeScope BeginLifetimeScope(object tag, Action<ContainerBuilder> co
         }
 
         /// <inheritdoc/>
-        public void AttachTrace(IResolvePipelineTracer tracer)
-        {
-            _rootLifetimeScope.AttachTrace(tracer);
-        }
+        public DiagnosticListener DiagnosticSource => this._rootLifetimeScope.DiagnosticSource;
```

Comment:
The `DiagnosticListener` implements `IDisposable`. Should we dispose it when the `Container` is disposed?

Reply:
Updated the `LifetimeScope` constructor such that if it is the thing creating the `DiagnosticSource` it also adds the source to its `DisposeTracker` so when the scope is disposed, the diagnostic source will also be disposed.